### PR TITLE
added user dtos, mock service, repository

### PR DESCRIPTION
**Pull request by IvanFadieiev**

### CODE REVIEW

File: src/main/java/com/jvavateam/carsharingapp/dto/user/UserRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.jvavateam.carsharingapp.dto.user;
+
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+
+public record UserRequestDto(@Email @NotBlank String email,
+                             @NotBlank @Size(min = 8) String password,
```
Comment (Dimagaa) [inline src/main/java/com/jvavateam/carsharingapp/dto/user/UserRequestDto.java]:
```suggestion
                             @NotBlank  @Pattern(regexp = "^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)[a-zA-Z\\d]{6,}$",
                                         message = "Minimum eight characters, at least one uppercase"
                                        + " letter, one lowercase letter and one number") 
                              String password,
```

**Review by Dimagaa — APPROVED** [review]
Well done! Consider using a pattern for password

File: src/main/java/com/jvavateam/carsharingapp/dto/user/UserRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.jvavateam.carsharingapp.dto.user;
+
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+
+public record UserRequestDto(@Email @NotBlank String email,
```
Comment (holsoni) [inline src/main/java/com/jvavateam/carsharingapp/dto/user/UserRequestDto.java:8]: I think that we can use that cross-field validation here, what if our password doesn't match repeated password?

**Review by holsoni — CHANGES_REQUESTED** [review]

File: src/main/java/com/jvavateam/carsharingapp/dto/user/UserRequestDto.java (hunk 2 above)
Comment (Dimagaa) [reply src/main/java/com/jvavateam/carsharingapp/dto/user/UserRequestDto.java:8]: @holsoni Cross-field validation will implemented as an extra task. I have already added it to the Trello

Comment (holsoni) [reply src/main/java/com/jvavateam/carsharingapp/dto/user/UserRequestDto.java:8]: Sorry, maybe i missed that

**Review by holsoni — APPROVED** [review]

**Pull request merged**